From 4247f7a0dcde71d8dff7f2b7a3ec195ecb11098c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 12 Mar 2024 00:17:46 +0800
Subject: [PATCH 2/6] AOSCOS: fix(iris_bufmgr.c): set PAGE_SIZE as 16384 on
 LoongArch (LP64)

16KiB is the most commonly found kernel page size on LoongArch
platforms. Limit this to LP64 platforms for now, as this is the only
one that we support.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/gallium/drivers/iris/iris_bufmgr.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gallium/drivers/iris/iris_bufmgr.c b/src/gallium/drivers/iris/iris_bufmgr.c
index fb59ea226e2..ae2ac3e083d 100644
--- a/src/gallium/drivers/iris/iris_bufmgr.c
+++ b/src/gallium/drivers/iris/iris_bufmgr.c
@@ -96,8 +96,16 @@
  * included here.
  */
 #ifndef PAGE_SIZE
+/* 16KiB is the most commonly found kernel page size on LoongArch
+ * platforms. Limit this to LP64 platforms for now, as this is the only
+ * one that we support.
+ */
+#if defined(__loongarch_lp64)
+#define PAGE_SIZE 16384
+#else
 #define PAGE_SIZE 4096
 #endif
+#endif
 
 #define WARN_ONCE(cond, fmt...) do {                            \
    if (unlikely(cond)) {                                        \
-- 
2.48.1

